src/Compilers/CSharp/Portable/Syntax/InternalSyntax/CSharpSyntaxNode.cs

```diff
@@ -270,36 +270,33 @@ public override bool IsTriviaWithEndOfLine()
         /// </remarks>
         public override SyntaxNode GetStructure(Microsoft.CodeAnalysis.SyntaxTrivia trivia)
         {
-            if (trivia.HasStructure)
+            if (!trivia.HasStructure)
             {
-                var parent = trivia.Token.Parent;
-                if (parent != null)
+                return null;
+            }
+
+            var parent = trivia.Token.Parent;
+            if (parent is null)
+            {
+                return CSharp.Syntax.StructuredTriviaSyntax.Create(trivia);
+            }
+
+            var structsInParent = s_structuresTable.GetValue(parent, static _ => new List<(CodeAnalysis.SyntaxTrivia, SyntaxNode)>(capacity: 1));
```

Comment:
alternatively, why not have this just be a map from parent to SmallDictionary?  SmallDict is alreayd teeny.  It's just an AVL tree.  So with a single key/value, it's just one root node.